services/stackexchange/stackexchange-base.spec.js

```diff
@@ -16,23 +31,47 @@ class DummyStackExchangeService extends StackExchangeBase {
   }
 }
 
-describe('StackExchangeBase', function () {
-  describe('auth', function () {
-    cleanUpNockAfterEach()
+// format is [class, example response from server]
+const testClasses = [
+  [DummyStackExchangeService, { message: 'fake message' }],
+  [StackExchangeMonthlyQuestions, { total: 8 }],
+  [StackExchangeReputation, { items: [{ reputation: 8 }] }],
+  [StackExchangeQuestions, { items: [{ count: 8 }] }],
+]
 
-    const config = { private: { stackapps_api_key: 'fake-key' } }
+for (const [serviceClass, dummyResponse] of testClasses) {
+  testAuth(serviceClass, dummyResponse)
+}
 
-    it('sends the auth information as configured', async function () {
-      const scope = nock('https://api.stackexchange.com')
-        .get('/2.2/tags/python/info')
-        .query({ key: 'fake-key' })
-        .reply(200, { message: 'fake message' })
+function testAuth(serviceClass, dummyResponse) {
+  describe(serviceClass.name, function () {
+    describe('auth', function () {
+      cleanUpNockAfterEach()
 
-      expect(
-        await DummyStackExchangeService.invoke(defaultContext, config, {}),
-      ).to.deep.equal({ message: 'fake message' })
+      const config = { private: { stackapps_api_key: 'fake-key' } }
+      const firstOpenapiPath = Object.keys(serviceClass.openApi)[0]
+      const exampleInvokeParams = serviceClass.openApi[
+        firstOpenapiPath
+      ].get.parameters.reduce((acc, obj) => {
+        acc[obj.name] = obj.example
+        return acc
+      }, {})
```

Comment:
Having read it, I can see what you're doing here but initially it wasn't obvious.
I think it would really help if we split these two statements out into a named function (maybe `getExampleCall()`?), write a quick comment explaining that we're using the documentation examples to extract a valid call we can make. Then this could actually go in https://github.com/badges/shields/blob/master/services/test-helpers.js because the same function would be applicable to any service. As we refactor more auth tests, we will want to use this everywhere.

Reply:
Its a great idea to break out this function and reuse the code at test-helpers. Will do that.
I will rework the function to make it as readable as i can.

Reply:
Fixed in https://github.com/badges/shields/pull/9681/commits/18ec38723a38bdbeca1edd7b6e6d057a361375ac